astropy/coordinates/tests/test_matching.py

```diff
@@ -153,98 +153,150 @@ def test_matching_method():
+            ICRS(ra=[] * u.deg, dec=[] * u.deg, distance=[] * u.kpc),
+            id="empty_catalog",
+        ),
+        pytest.param(
+            ICRS(ra=[] * u.deg, dec=[] * u.deg, distance=[] * u.kpc),
+            ICRS(ra=[] * u.deg, dec=[] * u.deg, distance=[] * u.kpc),
+            id="empty_both",
+        ),
+    ],
+)
+def test_search_around_empty_input(sources, catalog, function, search_limit):
     # Test when one or both of the coordinate arrays is empty, #4875
-    empty = ICRS(ra=[] * u.degree, dec=[] * u.degree, distance=[] * u.kpc)
-    idx1, idx2, d2d, d3d = search_around_sky(empty, coo2, 1 * u.arcsec)
-    assert idx1.size == idx2.size == d2d.size == d3d.size == 0
-    assert idx1.dtype == idx2.dtype == int
-    assert d2d.unit == u.deg
-    assert d3d.unit == u.kpc
-    idx1, idx2, d2d, d3d = search_around_sky(coo1, empty, 1 * u.arcsec)
-    assert idx1.size == idx2.size == d2d.size == d3d.size == 0
-    assert idx1.dtype == idx2.dtype == int
-    assert d2d.unit == u.deg
-    assert d3d.unit == u.kpc
-    empty = ICRS(ra=[] * u.degree, dec=[] * u.degree, distance=[] * u.kpc)
-    idx1, idx2, d2d, d3d = search_around_sky(empty, empty[:], 1 * u.arcsec)
-    assert idx1.size == idx2.size == d2d.size == d3d.size == 0
-    assert idx1.dtype == idx2.dtype == int
-    assert d2d.unit == u.deg
-    assert d3d.unit == u.kpc
-    idx1, idx2, d2d, d3d = search_around_3d(empty, coo2, 1 * u.m)
-    assert idx1.size == idx2.size == d2d.size == d3d.size == 0
-    assert idx1.dtype == idx2.dtype == int
-    assert d2d.unit == u.deg
-    assert d3d.unit == u.kpc
-    idx1, idx2, d2d, d3d = search_around_3d(coo1, empty, 1 * u.m)
-    assert idx1.size == idx2.size == d2d.size == d3d.size == 0
-    assert idx1.dtype == idx2.dtype == int
-    assert d2d.unit == u.deg
-    assert d3d.unit == u.kpc
-    idx1, idx2, d2d, d3d = search_around_3d(empty, empty[:], 1 * u.m)
-    assert idx1.size == idx2.size == d2d.size == d3d.size == 0
-    assert idx1.dtype == idx2.dtype == int
+    idx1, idx2, d2d, d3d = function(sources, catalog, search_limit)
+    assert idx1.size == 0
+    assert idx2.size == 0
+    assert d2d.size == 0
+    assert d3d.size == 0
+    assert idx1.dtype == int
+    assert idx2.dtype == int
     assert d2d.unit == u.deg
     assert d3d.unit == u.kpc
 
+
+@pytest.mark.parametrize(
+    "function,search_limit",
+    [
+        pytest.param(func, limit, id=func.__name__)
+        for func, limit in ([search_around_3d, 1 * u.m], [search_around_sky, 1 * u.deg])
+    ],
+)
+def test_search_around_no_dist_input_output_units(function, search_limit):
     # Test that input without distance units results in a
     # 'dimensionless_unscaled' unit
-    cempty = SkyCoord(ra=[], dec=[], unit=u.deg)
-    idx1, idx2, d2d, d3d = search_around_3d(cempty, cempty[:], 1 * u.m)
-    assert d2d.unit == u.deg
-    assert d3d.unit == u.dimensionless_unscaled
-    idx1, idx2, d2d, d3d = search_around_sky(cempty, cempty[:], 1 * u.m)
```

Comment:
The unit for the search limit was wrong, but it is correct in the new parametrization.